### PR TITLE
RYA-241 Adding GeoWave indexer to rya.geoindexing.

### PR DESCRIPTION
**Pull request by ejwhite922**
## Description
Added GeoWave indexer to rya.geoindexing.  This is a configurable option that can be used in place of GeoMesa.

### Tests
Unit Tests

### Links
[Jira](https://issues.apache.org/jira/browse/RYA-241)

### Checklist
- [x] Code Review
- [x] Squash Commits

#### People To Review
@pujav65 
@amihalik 
@isper3at 
@DLotts

### CODE REVIEW

File: extras/indexingExample/src/main/assembly/assembly.xml [hunk 1]
```diff
@@ -33,6 +33,7 @@ under the License.
             <includes>
                 <include>org.apache.rya:rya.indexing:*:accumulo-server</include>
                 <include>org.locationtech.geomesa:geomesa-accumulo-distributed-runtime:*</include>
+                <include>mil.nga.giat:geowave-datastore-accumulo:*</include>
```
Comment (amihalik) [inline extras/indexingExample/src/main/assembly/assembly.xml]: I think the dependency is something like `geowave-deploy-0.9.3-accumulo`?  I can't find this dependency on their maven repository, though

File: pom.xml [hunk 2]
```diff
@@ -80,6 +80,7 @@ under the License.
         <pig.version>0.9.2</pig.version> <!-- Newest: 0.15.0 -->
 
         <geomesa.version>1.3.0-m1</geomesa.version> <!-- Newest: 1.3.0-m1 -->
+        <geowave.version>0.9.4-SNAPSHOT</geowave.version> <!-- Newest: 0.9.4-SNAPSHOT -->
```
Comment (amihalik) [inline pom.xml]: Can you back this down to a released version (e.g. 0.9.3)?

File: extras/rya.geoindexing/src/main/java/org/apache/rya/indexing/accumulo/geo/GeoWaveGeoIndexer.java [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+
+        final Map<String, Serializable> params = new HashMap<>();
+        params.put("zookeeper", zookeepers);
+        params.put("instance", instanceId);
+        params.put("user", user);
+        params.put("password", password);
+        params.put("namespace", tableName);
+        params.put("gwNamespace", tablePrefix + getClass().getSimpleName());
+
+        params.put("Lock Management", LockManagementType.MEMORY.toString());
```
Comment (ejwhite922) [inline extras/rya.geoindexing/src/main/java/org/apache/rya/indexing/accumulo/geo/GeoWaveGeoIndexer.java]: It should handle multiple clients.

File: extras/indexingExample/src/main/assembly/assembly.xml (hunk 1 above)
Comment (ejwhite922) [inline extras/indexingExample/src/main/assembly/assembly.xml]: I added each geowave dependency individually.  The accumulo deploy dependency isn't available on the maven repo.  We'd have to build it if we wanted to use it.  The deploy pom is here: https://github.com/ngageoint/geowave/blob/9d6cfa9fb40e6e17650df3b41854d91402a91648/deploy/pom.xml

File: pom.xml (hunk 2 above)
Comment (ejwhite922) [inline pom.xml]: Changed to 0.9.3

**Comment by pujav65** [comment]
asfbot build

**Pull request closed without merging**